File: level/_240125/swea_1860.java
import java.util.Arrays;
import java.util.List;
import java.util.Scanner;

public class swea_1860 {

    public static int N;
    public static int M;
    public static int K;
    public static String answer;
    public static void main(String[] args){
        Scanner sc = new Scanner(System.in);

        int T = sc.nextInt();
        for(int test = 1 ; test < T +1 ; test++){
            N = sc.nextInt();
            M = sc.nextInt();
            K = sc.nextInt();
            answer = "Possible";

            int[] customers = new int[N];
            for(int i = 0 ; i < N ; i++){
                customers[i] = sc.nextInt();
            }

            Arrays.sort(customers);
            for(int i = 0 ; i < N ; i++){
                if((customers[i] / M) * K < i + 1){
                    answer = "Impossible";
                }
            }
            System.out.println("#" + test +" " + answer);
        }
    }
}

File: level/_240125/swea_1860.py
T = int(input())

for test_case in range(1, T+1):
    answer = "Possible"
    N, M, K = map(int, input().split())
    customer = list(map(int, input().split()))

    customer.sort()
    for i in range(N):
        if (customer[i] // M) * K  < i + 1:
            answer = "Impossible"
    print("#{} {}".format(test_case, answer))